client/src/containers/ProfileCreation/ProfileCreation1/index.js

```diff
@@ -0,0 +1,297 @@
+import React from 'react';
+import styled from 'styled-components';
+import RedAsterisk from '../../../components/redAsterisk/index';
+import SearchTag from '../../../components/searchTag/index';
+import Checkbox from '../../../components/checkbox/index';
+import Dropdown from '../../../components/dropdown/index';
+import TextInput from '../../../components/textInput/index';
+import GrayTag from '../../../components/grayTag/index';
+import { inputData, tagData } from './data';
+import GrayTagContainer from '../../../components/grayTag/container';
+
+const ProfileCreation1 = ({
+  clubProfile,
+  setClubProfile,
+  clubNameRef,
+  shortDescRef,
+  longDescRef,
+}) => {
+  const tagComponents = tagData.map((tagName, index) => {
+    return (
+      <TagContainer key={`tag-${index}`}>
+        <SearchTag
+          text={tagName}
+          data={clubProfile}
+          setData={setClubProfile}
+          objId='tags'
+          dataLimitSize={5}
+        />
+      </TagContainer>
+    );
+  });
+
+  const inputs = inputData.map((item, index) => {
+    const key = Object.keys(item)[0];
+    const getRef = () => {
+      if (key === 'clubName') {
+        console.log('clubName');
+        return clubNameRef;
+      } else if (key === 'shortDesc') return shortDescRef;
+      else if (key === 'longDesc') return longDescRef;
+    };
+    return (
+      <InputContainer key={`input-${index}`}>
+        <TextInput
+          compulsory
+          width='100%'
+          multiline={item[key].multiline}
+          height={item[key].height}
+          characterMax={item[key].characterMax}
+          labelHeader={item[key].labelHeader}
+          labelDesc={item[key].labelDesc}
+          identifier={key}
+          reference={getRef()}
+        />
+      </InputContainer>
+    );
+  });
+
+  return (
+    <StyledBody>
+      <Column left>
+        <TagHeader>
+          <RedAsterisk>*</RedAsterisk>
+          Select up to 5 tags that describe your club:
+        </TagHeader>
+        <TagsContainer>{tagComponents}</TagsContainer>
+      </Column>
+      <Column right>
+        {inputs}
+        <ClubSize clubProfile={clubProfile} setClubProfile={setClubProfile} />
+        <NewMembers clubProfile={clubProfile} setClubProfile={setClubProfile} />
+        <RequireApplication
+          clubProfile={clubProfile}
+          setClubProfile={setClubProfile}
+        />
+        <MeetFrequency
+          clubProfile={clubProfile}
+          setClubProfile={setClubProfile}
+        />
+      </Column>
+    </StyledBody>
+  );
+};
+
+export default ProfileCreation1;
```

Comment:
maybe separate this and the below code into different files? this gets pretty long

Reply:
but keep all file within the ProfileCreation Container?